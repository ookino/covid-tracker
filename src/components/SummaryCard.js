/* eslint-disable react/prop-types */
import globe from '../assets/globe.svg';

const SummaryCard = ({ summary }) => (
  <div className="card-summary">
    <img className="card-summary__img" src={globe} alt="globe" />
    <div className="card-summary__meta">
      <p className="card-summary__meta-p">{`${summary.cases} cases`}</p>
      <p className="card-summary__meta-p ">
        {summary.recovered}
        <span className="card-summary__meta-p--recovered"> recovered</span>
      </p>
      <p className="card-summary__meta-p  ">
        {summary.deaths}
        <span className="card-summary__meta-p--deaths"> deaths</span>
      </p>
    </div>
  </div>
);

export default SummaryCard;
