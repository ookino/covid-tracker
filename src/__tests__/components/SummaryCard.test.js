import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import SummaryCard from '../../components/SummaryCard';

it('renders SummaryCard Component correctly', () => {
  const summary = {
    cases: 43564675,
    recovered: 523464,
    deaths: 23543,
  };
  const tree = renderer
    .create(
      <Router>
        <SummaryCard summary={summary} />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
