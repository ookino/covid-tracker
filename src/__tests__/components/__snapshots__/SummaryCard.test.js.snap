// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SummaryCard Component correctly 1`] = `
<div
  className="card-summary"
>
  <img
    alt="globe"
    className="card-summary__img"
    src="globe.svg"
  />
  <div
    className="card-summary__meta"
  >
    <p
      className="card-summary__meta-p"
    >
      43564675 cases
    </p>
    <p
      className="card-summary__meta-p "
    >
      523464
      <span
        className="card-summary__meta-p--recovered"
      >
         recovered
      </span>
    </p>
    <p
      className="card-summary__meta-p  "
    >
      23543
      <span
        className="card-summary__meta-p--deaths"
      >
         deaths
      </span>
    </p>
  </div>
</div>
`;
